src/extractor/edge_based_graph_factory.cpp

```diff
@@ -208,6 +208,9 @@ void EdgeBasedGraphFactory::Run(ScriptingEnvironment &scripting_environment,
 /// Returns the number of edge based nodes.
 unsigned EdgeBasedGraphFactory::RenumberEdges()
 {
+    // heuristic: node-based graph node is a simple intersection with four edges (edge-based nodes)
+    m_edge_based_node_weights.reserve(4 * m_node_based_graph->GetNumberOfNodes());
```

Comment:
just checked: after edges zipping  `GetNumberOfEdges` overestimates the  number valid edges, so i would keep 4 * #nodes to avoid  regression in memory consumption